[OSDOCS#18857]: Added a new module for Fencing Validator script (#110148)
openshift/openshift-docs: modules/fencing-validator-script-for-disruptive-checks.adoc

> :_mod-docs-content-type: CONCEPT

Reviewer: Metatdata missing at the start of the file to indicate what assembly the file belongs to. Example: 

```
// Module included in the following assemblies:
//
// * networking/configuring_ingress_cluster_traffic/configuring-ingress-cluster-traffic-nodeport.adoc

:_mod-docs-content-type: CONCEPT
[id="nw-using-nodeport_{context}"]
= Using a NodePort to get traffic into the cluster
```

Reviewer: File name: modules/Installation-fencing-validator-disruptive-tests.adoc / modules/installation-fencing-validator-disruptive-tests.adoc

Ideally the filename and ID should match to make for easier xref management.

[source,terminal]
----
$ pcs stonith fence <node>
----

You can check whether Two-Node with Fencing (TNF) cluster setup actually works by running the following command: 

[source,terminal]
----
$ oc debug node/<node-name> --chroot /host /usr/local/bin/fencing_validator --disruptive
----

[WARNING]
====
The `--disruptive` flag fences each control-plane node one at a time and verifies recovery. The flag causes temporary workload disruption. Only use this during maintenance windows or initial validation. The `--disruptive` flag actually performs the Shoot The Other Node In The Head (STONITH) fence operations such as power cycle or VM reset, and not a graceful shutdown.
====

The fencing validator script with `--disruptive` flag runs the following checks: 

. **Fence Node A** - Triggers STONITH to reboot the first control-plane node.
. **Verify NotReady** - Waits for Kubernetes to report the node as `NotReady`, which confirms the reboot happened.
. **Verify recovery** - Waits for the node to come back to the `Ready` state, rejoin the Pacemaker cluster as `ONLINE`, and for etcd to regain quorum.
. **Post-recovery daemon check** - Re-checks all daemons are healthy after recovery.

> . **Fence Node B** - Repeats the same process for the second node.

Reviewer 2: ```suggestion
. **Fence Node B** - Triggers STONITH to reboot the second node.
```

> . **Verify NotReady, recovery, and daemons** - Same recovery validation for Node B.

Reviewer 2: this is listed separately for Node A but combines for Node B. You can split them.

//Module included in the following assemblies:
//
// *installing/installing_two_node_cluster/installing_tnf/install-post-tnf.adoc

:_mod-docs-content-type: CONCEPT
[id="fencing-validator-script-for-disruptive-checks_{context}"]
= Fencing validator script for disruptive checks

[role="_abstract"]
You can validate your cluster's resilience and perform disruptive checks from a peer node by using the `fencing_validator` script. By executing these simulated failures, you can ensure your high-availability environment correctly isolates and recovers from errors.

[source,terminal]
----

> $ pcs stonith fence <node>

Reviewer 3: Introduce this command, like you did on line 17 below.

----

> You can check whether Two-Node with Fencing (TNF) cluster setup actually works by running the following command: 

Reviewer 3: Consideration -- maybe a better word than works? 

Not sure if an article is needed before "Two-Node Fencing". You used it in the assembly so I'm suggesting it here. 

```suggestion
You can check whether a Two-Node with Fencing (TNF) cluster setup operates correctly by running the following command: 
```

[source,terminal]
----
$ oc debug node/<node-name> --chroot /host /usr/local/bin/fencing_validator --disruptive
----

[WARNING]
====

> The `--disruptive` flag fences each control-plane node one at a time and verifies recovery. The flag causes temporary workload disruption. Only use this during maintenance windows or initial validation. The `--disruptive` flag actually performs the Shoot The Other Node In The Head (STONITH) fence operations such as power cycle or VM reset, and not a graceful shutdown.

Reviewer 3: Reorganizing this a little bit as a suggestion:

```suggestion
The `--disruptive` flag fences each control-plane node one at a time and verifies recovery.  The `--disruptive` flag performs the Shoot The Other Node In The Head (STONITH) fence operations such as power cycle or VM reset. It does not perform a graceful shutdown, and causes temporary workload disruption. 
```
